crates/fm/src/lib.rs

```diff
@@ -38,12 +38,21 @@ pub struct FileManager {
 impl FileManager {
     // XXX: Maybe use a AsRef<Path> here, for API ergonomics
     pub fn add_file(&mut self, path_to_file: &Path, file_type: FileType) -> Option<FileId> {
-        let source = file_reader::read_file_to_string(path_to_file).ok()?;
+        // Handle both relative file paths and std/lib virtual paths.
+        let base = Path::new(".").canonicalize().expect("Base path canonicalize failed"); // Should never fail
```

Comment:
```suggestion
        let base = Path::new(".").canonicalize().expect("Base path canonicalize failed");
```
We can remove the comment, we should assume any call to `expect` shouldn't fail. If we expect it to fail, it should be a user-facing error message with source location, etc.